src/components/text-editor/examples/text-editor-with-inline-images-file-storage.tsx

```diff
@@ -109,19 +113,44 @@ export class TextEditorWithInlineImagesExample {
         }
     };
 
-    private handleImageRemoved = (
-        event: LimelTextEditorCustomEvent<ImageInfo>,
+    private handleMetadataChange = (
+        event: LimelTextEditorCustomEvent<MetadataInfo>,
     ) => {
-        const imageInfo = event.detail;
-        console.log(`Image deleted: ${imageInfo.fileInfoId}`);
+        const removedImages = this.getRemovedImages(
+            this.metadata,
+            event.detail,
+        );
+
+        removedImages.forEach((image) => {
+            console.log(`Image removed: ${image.fileInfoId}`);
+
+            if (image.state === ImageState.SUCCESS) {
+                this.removeImage(image);
+            }
+        });
+
+        this.metadata = event.detail;
+    };
+
+    private getRemovedImages(
+        oldMetadata: MetadataInfo,
+        newMetadata: MetadataInfo,
+    ): ImageInfo[] {
+        return oldMetadata.images.filter(
+            (oldImage) =>
+                !newMetadata.images.some(
+                    (newImage) => newImage.fileInfoId === oldImage.fileInfoId,
+                ),
+        );
```

Comment:
Maybe better to avoid nested loop ? And using a `Set` for faster lookups, improving performance when comparing
```suggestion
         const newImageIds = new Set(
            newMetadata.images.map((image) => image.fileInfoId),
        );

        return oldMetadata.images.filter(
            (image) => !newImageIds.has(image.fileInfoId),
        );
```

Reply:
⚡ [4ffe955](https://github.com/Lundalogik/lime-elements/pull/3547/commits/4ffe955c7c15cc64e482061680bb8126562429ef)